src/server/pbsd_init.c

```diff
@@ -668,58 +663,55 @@ pbsd_init(int type)
 
 	/* 4. Check License information */
 
-	init_fl_license_attrs(&licenses);
+	reset_license_counters(&license_counts);
 
 	fd = open(path_usedlicenses, O_RDONLY, 0400);
 
 	if ((fd == -1) ||
-		(read(fd, &usedlicenses, sizeof(usedlicenses)) !=
-		sizeof(usedlicenses))) {
-		usedlicenses.lu_max_hr      = 0;
-		usedlicenses.lu_max_day     = 0;
-		usedlicenses.lu_max_month   = 0;
-		usedlicenses.lu_max_forever = 0;
+		(read(fd, &(license_counts.licenses_high_use), sizeof(pbs_licenses_high_use)) !=
```

Comment:
Make it a function and keep as part of the library if possible

Reply:
Make it part of function initialize_lic_counts()  - will stucture the code better.